### ISSUE TITLE
Question about mass insertion

### ISSUE THREAD

**Issue by kimtree** [mediocregopher/radix#158]
Hi guys,

I'm trying to use mass insertion which described here. https://redis.io/topics/mass-insert
Can I use Radix to send these mass insertion pipe command to Redis server?

**Comment by mediocregopher** [comment]
Hi again! Yeah that should definitely be possible, and you have a couple of options on how to do it:

* For each `CmdAction` call `MarshalRESP`, passing in an `*os.File`. Once all commands are written to the file you can use the `redis-cli --pipe` as described in that link.

* Same as above, but rather than writing to a file you can write directly to `os.Stdout`, and write directly to `redis-cli`, like: `./bulk-insert | redis-cli --pipe`.

* Basically implement something similar to `redis-cli --pipe` in go. Have one go-routine writing `CmdAction`s to a `radix.Conn` using it's `Encode` method, ending by `ECHO`ing a random string. At the same time, have another go-routine calling `Decode` on the `Conn`, looking for that random string.

So you see, there's lots of options :) hope this helps!

**Comment by kimtree** [comment]
@mediocregopher Oh, Such kindness answers! Thank you so much. This helps a lot 👍
